DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/devlink/test_devlink_single_block.py

```diff
@@ -0,0 +1,412 @@
+         {'ixp': tg_ports[1], 'ip': '2.2.2.4', 'gw': '2.2.2.1', 'plen': 24}]
+    )
+
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, dut_ports, dev_groups)
+
+    # 2.Create an ingress qdisc on first port connected to Ixia and add a rule with (l2, l3, l4) selectors
+    out = await TcQdisc.add(
+        input_data=[{dev_name: [
+            {'dev': dut_ports[0], 'direction': 'ingress'}]}])
+    err_msg = f'Verify no error on setting Qdist for {dut_ports[0]}.\n{out}'
+    assert out[0][dev_name]['rc'] == 0, err_msg
+
+    rule = {'action': {'trap': '',
+                       'police': {'rate': f'{policer_rate}bps', 'burst': policer_rate + 1000,
+                                  'conform-exceed': '', 'drop': ''}},
+            'want_ip': True if traffic in ['l3', 'l4'] else False,
+            'want_port': True if traffic == 'l4' else False,
+            'skip_sw': True,
+            'want_mac': True if traffic == 'l2' else False,
+            'want_vlan': True if traffic == 'l2' else False,
+            'want_tcp': choice([True, False]) if traffic == 'l4' else False,
+            'want_vlan_ethtype': False,
+            }
+
+    tc_rule = tcutil_generate_rule_with_random_selectors(dut_ports[0], **rule)
+
+    if traffic == 'l2':
+        tc_rule['protocol'] = '0x8100'
+        tc_rule['filtertype']['src_mac'] = 'aa:aa:aa:aa:aa:00'
+        tc_rule['filtertype']['dst_mac'] = 'aa:00:00:00:00:55'
+    elif traffic == 'l3':
+        tc_rule['filtertype']['src_mac'] = '00:00:00:00:00:01'
+        tc_rule['filtertype']['dst_mac'] = '00:00:00:00:00:02'
+    elif traffic == 'l4':
+        tc_rule['filtertype']['src_mac'] = '00:00:00:00:00:01'
+        tc_rule['filtertype']['dst_mac'] = '00:00:00:00:00:02'
+        tc_rule['filtertype']['src_ip'] = '1.1.1.2'
+        tc_rule['filtertype']['dst_ip'] = '2.2.2.3'
+
+    out = await TcFilter.add(input_data=[{dev_name: [tc_rule]}])
+    assert out[0][dev_name]['rc'] == 0, f'Failed to create tc rule \n{out}'
+
+    # 3.Prepare a matching stream for the randomly selected selectors and transmit traffic
+    out = await TcFilter.show(input_data=[{dev_name: [
+        {'dev': dut_ports[0], 'direction': 'ingress', 'options': '-j'}]}], parse_output=True)
+    assert out[0][dev_name]['rc'] == 0, f'Failed to get tc rule \n{out}'
+
+    streams = tcutil_tc_rules_to_tgen_streams({dut_ports[0]: out[0][dev_name]['parsed_output']},
+                                              frame_rate_pps=100, frame_rate_type='line_rate',
+                                              frame_size=frame_size)
+
+    await tgen_utils_setup_streams(tgen_dev, config_file_name=None, streams=streams)
+    await tgen_utils_start_traffic(tgen_dev)
+
+    # 4.Verify CPU trapped packet rate is as expected
+    # Calculate expected pkt rate based on policer rate limitation
+    exp_rate = policer_rate / frame_size
+    await asyncio.sleep(15)
+
+    actual_rate = await get_cpu_traps_rate_code_avg(dent_dev, '195', 'sw')
```

Comment:
please add a variable for the cpu code, so it is clear for what type of traffic you are calculating rate.